### PR TITLE
feat: filtering

### PR DESCRIPTION
**Pull request by SuperMarcomen**

### CODE REVIEW

File: server/src/main/java/de/uftos/controller/ConstraintController.java [hunk 1]
```diff
@@ -30,7 +31,8 @@ public ConstraintController(ConstraintSignatureService constraintSignatureServic
   }
 
   @GetMapping()
-  public Page<ConstraintSignature> getConstraintSignatures(Pageable pageable) {
+  public Page<ConstraintSignature> getConstraintSignatures(Pageable pageable,
+                                                           Optional<String> name) {
     return this.constraintSignatureService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/ConstraintController.java:36]:
```suggestion
    return this.constraintSignatureService.get(pageable, name);
```

File: server/src/main/java/de/uftos/controller/ConstraintController.java [hunk 2]
```diff
@@ -47,7 +49,8 @@ public ConstraintInstance createConstraintInstance(@PathVariable String id,
 
   @GetMapping("/{id}/instances")
   public Page<ConstraintInstance> getConstraintInstances(@PathVariable String id,
-                                                         Pageable pageable) {
+                                                         Pageable pageable,
+                                                         Optional<String> argument) {
     return this.constraintInstanceService.get(id, pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/ConstraintController.java:54]:
```suggestion
    return this.constraintInstanceService.get(id, pageable, argument);
```

File: server/src/main/java/de/uftos/controller/GradeController.java [hunk 3]
```diff
@@ -56,7 +57,7 @@ public GradeResponseDto createGrade(@RequestBody GradeRequestDto grade) {
    * @return the page of grades fitting the parameters.
    */
   @GetMapping()
-  public Page<GradeResponseDto> getGrades(Pageable pageable) {
+  public Page<GradeResponseDto> getGrades(Pageable pageable, Optional<String> name) {
     return this.gradeService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/GradeController.java:61]:
```suggestion
    return this.gradeService.get(pageable, name);
```

File: server/src/main/java/de/uftos/controller/RoomController.java [hunk 4]
```diff
@@ -56,7 +57,9 @@ public Room createRoom(@RequestBody RoomRequestDto room) {
    * @return the page of rooms fitting the parameters.
    */
   @GetMapping()
-  public Page<Room> getRooms(Pageable pageable) {
+  public Page<Room> getRooms(Pageable pageable, Optional<String> name,
+                             Optional<String> buildingName, Optional<Integer> capacity,
+                             Optional<String[]> tags) {
     return this.roomService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/RoomController.java:63]:
```suggestion
    return this.roomService.get(pageable, name, buildingName, capacity, tags);
```

File: server/src/main/java/de/uftos/controller/StudentController.java [hunk 5]
```diff
@@ -54,7 +55,8 @@ public Student createStudent(@RequestBody StudentRequestDto student) {
    * @return the page of students fitting the parameters.
    */
   @GetMapping()
-  public Page<Student> getStudents(Pageable pageable) {
+  public Page<Student> getStudents(Pageable pageable, Optional<String> firstName,
+                                   Optional<String> lastName, Optional<String[]> tags) {
     return this.studentService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/StudentController.java:60]:
```suggestion
    return this.studentService.get(pageable, firstName, lastName, tags);
```

File: server/src/main/java/de/uftos/controller/StudentGroupController.java [hunk 6]
```diff
@@ -56,7 +57,7 @@ public StudentGroup createStudentGroup(@RequestBody StudentGroupRequestDto stude
    * @return the page of student groups fitting the parameters.
    */
   @GetMapping()
-  public Page<StudentGroup> getStudentGroups(Pageable pageable) {
+  public Page<StudentGroup> getStudentGroups(Pageable pageable, Optional<String> name) {
     return this.studentGroupService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/StudentGroupController.java:61]:
```suggestion
    return this.studentGroupService.get(pageable, name);
```

File: server/src/main/java/de/uftos/controller/SubjectController.java [hunk 7]
```diff
@@ -54,7 +55,7 @@ public Subject createSubject(@RequestBody SubjectRequestDto subject) {
    * @return the page of subjects fitting the parameters.
    */
   @GetMapping()
-  public Page<Subject> getSubjects(Pageable pageable) {
+  public Page<Subject> getSubjects(Pageable pageable, Optional<String> name) {
     return this.subjectService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/SubjectController.java:59]:
```suggestion
    return this.subjectService.get(pageable, name);
```

File: server/src/main/java/de/uftos/controller/TagController.java [hunk 8]
```diff
@@ -54,7 +55,7 @@ public Tag createTag(@RequestBody TagRequestDto tag) {
    * @return the page of tags fitting the parameters.
    */
   @GetMapping()
-  public Page<Tag> getTags(Pageable pageable) {
+  public Page<Tag> getTags(Pageable pageable, Optional<String> name) {
     return this.tagsService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/TagController.java:59]:
```suggestion
    return this.tagsService.get(pageable, name);
```

File: server/src/main/java/de/uftos/controller/TeacherController.java [hunk 9]
```diff
@@ -56,7 +57,9 @@ public Teacher createTeacher(@RequestBody TeacherRequestDto teacher) {
    * @return the page of teachers fitting the parameters.
    */
   @GetMapping()
-  public Page<Teacher> getTeachers(Pageable pageable) {
+  public Page<Teacher> getTeachers(Pageable pageable, Optional<String> firstName,
+                                   Optional<String> lastName, Optional<String> acronym,
+                                   Optional<String[]> subjects, Optional<String[]> tags) {
     return this.teacherService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/TeacherController.java:63]:
```suggestion
    return this.teacherService.get(pageable, firstName, lastName, acronym, subjects, tags);
```

File: server/src/main/java/de/uftos/controller/TimetableController.java [hunk 10]
```diff
@@ -53,7 +54,7 @@ public Timetable createTimetable(@RequestBody TimetableRequestDto timetable) {
    * @return the page of timetable fitting the parameters.
    */
   @GetMapping()
-  public Page<Timetable> getTimetables(Pageable pageable) {
+  public Page<Timetable> getTimetables(Pageable pageable, Optional<String> name) {
     return this.timetablesService.get(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/TimetableController.java:58]:
```suggestion
    return this.timetablesService.get(pageable, name);
```

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java [hunk 11]
```diff
@@ -45,7 +46,8 @@ public ConstraintInstance create(String signatureId, ConstraintInstanceRequestDt
    * @param pageable    contains the parameters for the page.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<ConstraintInstance> get(String signatureId, Pageable pageable) {
+  public Page<ConstraintInstance> get(String signatureId, Pageable pageable,
+                                      Optional<String> argument) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/ConstraintInstanceService.java:51]:
```suggestion
    return this.repository.findAll(pageable, argument);
```

File: server/src/main/java/de/uftos/services/ConstraintSignatureService.java [hunk 12]
```diff
@@ -30,9 +31,10 @@ public ConstraintSignatureService(ConstraintSignatureRepository repository) {
    * Gets a page of entries of the constraintSignature entity.
    *
    * @param pageable contains the parameters for the page.
+   * @param name     the name filter.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<ConstraintSignature> get(Pageable pageable) {
+  public Page<ConstraintSignature> get(Pageable pageable, Optional<String> name) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/ConstraintSignatureService.java:38]:
```suggestion
    return this.repository.findAll(pageable, name);
```

File: server/src/main/java/de/uftos/services/GradeService.java [hunk 13]
```diff
@@ -39,9 +40,10 @@ public GradeService(GradeRepository repository) {
    * Gets a page of entries of the grade table.
    *
    * @param pageable contains the parameters for the page.
+   * @param name     the name filter.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<GradeResponseDto> get(Pageable pageable) {
+  public Page<GradeResponseDto> get(Pageable pageable, Optional<String> name) {
     Page<Grade> grades = this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/GradeService.java:47]:
```suggestion
    Page<Grade> grades = this.repository.findAll(pageable, name);
```

File: server/src/main/java/de/uftos/services/RoomService.java [hunk 14]
```diff
@@ -33,10 +33,16 @@ public RoomService(RoomRepository repository) {
...
+   * @param buildingName the building name filter.
+   * @param capacity     the capacity filter.
+   * @param tags         the tags filter.
    * @return the page of entries fitting the parameters.
    */
-  public Page<Room> get(Pageable pageable) {
+  public Page<Room> get(Pageable pageable, Optional<String> name,
+                        Optional<String> buildingName, Optional<Integer> capacity,
+                        Optional<String[]> tags) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/RoomService.java:46]:
```suggestion
    return this.repository.findAll(pageable, name, buildingName, capacity, tags);
```

File: server/src/main/java/de/uftos/services/StudentGroupService.java [hunk 15]
```diff
@@ -33,9 +34,10 @@ public StudentGroupService(StudentGroupRepository repository) {
    * Gets a page of entries of the student group table.
    *
    * @param pageable contains the parameters for the page.
+   * @param name     the name filter.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<StudentGroup> get(Pageable pageable) {
+  public Page<StudentGroup> get(Pageable pageable, Optional<String> name) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/StudentGroupService.java:41]:
```suggestion
    return this.repository.findAll(pageable, name);
```

File: server/src/main/java/de/uftos/services/StudentService.java [hunk 16]
```diff
@@ -31,10 +31,14 @@ public StudentService(StudentRepository repository) {
...
+   * @param pageable  contains the parameters for the page.
+   * @param firstName the first name filter.
+   * @param lastName  the last name filter.
+   * @param tags      the tags filter.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<Student> get(Pageable pageable) {
+  public Page<Student> get(Pageable pageable, Optional<String> firstName,
+                           Optional<String> lastName, Optional<String[]> tags) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/StudentService.java:42]:
```suggestion
    return this.repository.findAll(pageable, lastName, tags);
```

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java (hunk 11 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ConstraintInstanceService.java:51]: That's not how this works. Also, that's logic, so let's skip it for now

File: server/src/main/java/de/uftos/services/StudentService.java (hunk 16 above)
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/StudentService.java:42]:
```suggestion
    return this.repository.findAll(pageable, firstName, lastName, tags);
```

File: server/src/main/java/de/uftos/services/ConstraintSignatureService.java (hunk 12 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ConstraintSignatureService.java:38]: Same here

File: server/src/main/java/de/uftos/services/GradeService.java (hunk 13 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/GradeService.java:47]: And here

File: server/src/main/java/de/uftos/services/RoomService.java (hunk 14 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/RoomService.java:46]: And here

File: server/src/main/java/de/uftos/services/StudentGroupService.java (hunk 15 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/StudentGroupService.java:41]: And here

File: server/src/main/java/de/uftos/services/SubjectService.java [hunk 17]
```diff
@@ -32,9 +32,10 @@ public SubjectService(SubjectRepository repository) {
    * Gets a page of entries of the subject table.
    *
    * @param pageable contains the parameters for the page.
+   * @param name the name filter.
    * @return the page of the entries fitting the parameters.
    */
-  public Page<Subject> get(Pageable pageable) {
+  public Page<Subject> get(Pageable pageable, Optional<String> name) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/SubjectService.java:39]:
```suggestion
    return this.repository.findAll(pageable, name);
```

File: server/src/main/java/de/uftos/services/TagService.java [hunk 18]
```diff
@@ -31,9 +32,10 @@ public TagService(TagRepository repository) {
    * Gets a page of entries of the tag table.
    *
    * @param pageable contains the parameters for the page.
+   * @param name     the name filter.
    * @return the page of entries fitting the parameters.
    */
-  public Page<Tag> get(Pageable pageable) {
+  public Page<Tag> get(Pageable pageable, Optional<String> name) {
     return this.repository.findAll(pageable);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/TagService.java:39]:
```suggestion
    return this.repository.findAll(pageable, name);
```

File: server/src/main/java/de/uftos/controller/TeacherController.java [hunk 19]
```diff
@@ -48,16 +49,33 @@ public Teacher createTeacher(@RequestBody TeacherRequestDto teacher) {
     return this.teacherService.create(teacher);
   }
 
-  /**
+  /*
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/TeacherController.java:52]: outcommented javadoc

File: server/src/main/java/de/uftos/services/TeacherService.java [hunk 20]
```diff
@@ -36,7 +36,21 @@ public TeacherService(TeacherRepository repository) {
    * @param pageable contains the parameters for the page.
    * @return the page of entries fitting the parameters.
    */
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/TeacherService.java:38]: Old JavaDoc still above

**Review by Lama64 — APPROVED** [review]

**Review by danieldietzler — APPROVED** [review]

**Pull request merged**